File: 897 - Increasing Order Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
let last = null;
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var increasingBST = function(root) {
    let t = new TreeNode();
    last = t;
    traverse(root);
    return t.right;
};

function traverse(root) {
    if (root == null) return;
    traverse(root.left);
    last.right = root;
    root.left = null;
    last = root;
    traverse(root.right);
}

File: 897 - Increasing Order Search Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    res = None
    def increasingBST(self, root: TreeNode) -> TreeNode:
        t = TreeNode()
        self.res = t
        self.traverse(root)
        return t.right

    def traverse(self, root):
        if root is None: return
        self.traverse(root.left)
        self.res.right = root
        root.left = None
        self.res = root
        self.traverse(root.right)
        
    